File: April/Abiji-2020/C++/Number of Students Unable to eat lunch/Number of Students unable to eat lunch.cpp
class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        vector<int> count(2,0);
        for(int sud: students){
            count[sud]++;
        }
        for(int i =0;i<sandwiches.size();i++){
            if (count[sandwiches[i]]==0) return sandwiches.size()-i;
            count[sandwiches[i]]--;
        }
        
        return 0;
    }

};

File: April/Abiji-2020/Golang /Number of Students Unable to eat Lunch/Number of students unable to eat lunch.go
func countStudents(students []int, sandwiches []int) int {
    
    count := [2]int{}
    
    for _, student := range students {
        count[student]++
    }

    for i, sandwich := range sandwiches {
        if count[sandwich] == 0 {
            return len(sandwiches) - i
        }
        
        count[sandwich]--
    }

    return 0
}
